packages/mdl-icon-toggle/README.md

```diff
@@ -0,0 +1,199 @@
+un-styled content.
+
+### Using with Font Awesome and similar libraries
+
+Font Awesome - as well as other popular icon font libraries - use their element's pseudo-elements
+in order to provide the icon, via the `content` property. However, MDL uses pseudo-elements for
+ripple styles. In order to get around this, you can nest the icon itself inside the icon toggle.
+
+```html
+<span class="mdl-icon-toggle" role="button" aria-pressed="false"
+      aria-label="Star this item" tabindex="0"
+      data-icon-inner-selector=".fa"
+      data-toggle-on='{"cssClass": "fa-star", "label": "Unstar this item"}'
+      data-toggle-off='{"cssClass": "fa-star-o", "label": "Star this item"}'>
+  <i class="fa fa-star-o" aria-hidden="true"></i>
+</span>
+```
+
+`data-icon-inner-selector` tells MDLIconToggle to look for an element within itself that matches
+that selector, and treat it as the element containing the icon. Also note the `aria-hidden`
+attribute on the icon. This is important to ensure that screen readers produce the correct output
+when reading this element.
+
+### Configuring the icon toggle states
+
+Note the use of `data-toggle-on` and `data-toggle-off` in the above examples. When an MDLIconToggle
+instance is toggled, it looks at this data to determine how to update the element. This is what
+allows MDLIconToggle to be so flexible. The `data-toggle-on` configuration will be used when the is
+MDLIconToggle is toggled on, and vice versa for `data-toggle-off`. Both data attributes are encoded
+as JSON and can contain the following properties:
+
+| Property | Description |
+| --- | --- |
+| `label` | The value to apply to the element's "aria-label" attribute. |
+| `content` | The text content to set on the element. Note that if an inner icon is used, the text content will be set on that element instead. |
+| `cssClass` | A css class to apply to the icon element for the given toggle state. The same rules regarding inner icon element's described for `content` apply here as well. |
+
+### Disabled icon toggles
+
+```html
+<i class="material-icon mdl-icon-toggle mdl-icon-toggle--disabled"
+   role="button" tabindex="-1" aria-pressed="false" aria-disabled="true"
+   data-toggle-on='{"content": "favorite"}' data-toggle-off='{"content": "favorite_border"}'></i>
+```
+
+### Theming
+
+`mdl-icon-toggle` ships with two css classes, `mdl-icon-toggle--primary` and
+`mdl-icon-toggle--accent` that allow you to color mdl-icon-toggle based on your primary and accent
+colors, respectively.
+
+### Listening for change events
+
+`MDLIconToggle` emits an `MDLIconToggle:change` custom event when the value of the icon toggle
+changes _as a result of user input_. This decision was made to align with how `change` events work
+for normal inputs. In addition, these events do not bubble and cannot be cancelled.
+
+The custom event's `detail` object contains a property `isOn` denoting whether or not the ripple is
```

Comment:
Yup totally right. That 100% reflects the status of the icon toggle. Not sure why I wrote that...great catch!